### PR TITLE
Remove dependency on doctrine/common

### LINKED ISSUE #3176: Eliminate dependency on doctrine/common
There's currently hard dependency on `doctrine/common` which pulls unnecessary packages like Lexer or Annotations.
What is actually used:
`Doctrine\Common\Cache` -> to be replaced by `doctrine/cache`
`Doctrine\Common\EventManager` -> needs to be extracted into `doctrine/event-manager`
`Doctrine\Common\Util\Debug` -> move into `doctrine/dbal` / extract to `doctrine/debug` / replace by something else?
`Doctrine\Common\Proxy\Exception\UnexpectedValueException` thrown from `SQLAnywhere16Platform` - this looks like a bug (wrong import)

Blocker for https://github.com/doctrine/migrations/issues/722

### PR DESCRIPTION
**Pull request by Majkl578**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues |  #3176

#### Summary

Replace doctrine/common with dependency on doctrine/cache + doctrine/event-manager.

This also drops indirect dependency on:
* doctrine/lexer
* doctrine/collections
* doctrine/annotations
* doctrine/inflector

closes  #3176

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "require": {
         "php": "^7.1",
         "ext-pdo": "*",
-        "doctrine/common": "^2.7.1"
+        "doctrine/cache": "^1.0",
+        "doctrine/event-manager": "^1.0@dev"
```
Comment (morozov) [inline composer.json]: Should we release this first?

Comment (Majkl578) [reply composer.json]: Depends, leaving the decision up to you. 😛 We should definitely have 1.0.0 before DBAL 2.8 either way.

Comment (Majkl578) [reply composer.json]: Bumped to stable. :+1:

**Comment by jwage** [comment]
If this is not considered a BC break, I think we should release this in a Doctrine DBAL 1.7.x release. @Ocramius thoughts?

**Comment by morozov** [comment]
I have a gut feeling it should be `2.8.0` rather than `2.7.x`. If it was a BC break, it would have to be `3.0.0`. If it was just a patch or a bug fix, it would be `2.7.x`. But it's more of an improvement and code reorganization.

**Review by morozov — APPROVED** [review]

**Comment by morozov** [comment]
Waiting for the consensus before merging (the upgrade file specifically say the change will have happened in `2.8.0`).

**Comment by Ocramius** [comment]
> If this is not considered a BC break, I think we should release this in a Doctrine DBAL 1.7.x release. 
 > @Ocramius thoughts?

Patch releases MUST leave dependencies un-touched, because every older system MUST be able to run a `composer update` at any time in order to receive eventual security patches, so the constraints are very important there.

We can make this land in `2.8`, but not `2.7`.

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
:ship:

**Pull request merged**